### PR TITLE
Backend Unit Tests via CI & Run tests in parallel

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

It was a hard struggle, but I win

image

See: https://github.com/Ocelot-Social-Community/Ocelot-Social/actions/runs/523898450

- Backend Tests via CI
- Run CI Tasks in parallel if possible
- Docker-Tests Builds

### Issues

- fixes https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4173

### Todo

- [x] Make Ticket to implement CI Cypress Tests

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,31 +1,261 @@
...
+        with:
+          name: docker-backend-test
+          path: /tmp
+      - name: Load Docker Image
+        run: docker load < /tmp/backend.tar
+      ##########################################################################
+      # LINT BACKEND ###########################################################
+      ##########################################################################
+      - name: backend | Lint
+        run: docker run --rm ocelotsocialnetwork/backend:test yarn run lint
```
Comment (Mogge) [inline .github/workflows/ci.yml:133]: Do we need a build to tun this? Why is it not possible to run eslint without a build?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,31 +1,261 @@
...
+        with:
+          name: docker-webapp-test
+          path: /tmp
+      - name: Load Docker Image
+        run: docker load < /tmp/webapp.tar
+      ##########################################################################
+      # LINT WEBAPP ############################################################
+      ##########################################################################
+      - name: webapp | Lint
+        run: docker run --rm ocelotsocialnetwork/webapp:test yarn run lint
```
Comment (Mogge) [inline .github/workflows/ci.yml:162]: Same here

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,31 +1,261 @@
...
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-backend-test
+          path: /tmp
+      - name: Load Docker Image
+        run: docker load < /tmp/backend.tar
+      ##########################################################################
+      # UNIT TESTS BACKEND #####################################################
+      ##########################################################################
+      # TODO: Why do we need those .envs?
```
Comment (Mogge) [inline .github/workflows/ci.yml:197]: Looking at the .envs in webapp and backend says to me that we do not need the .env form the webapp here.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,31 +1,261 @@
...
+          path: /tmp
+      - name: Load Docker Image
+        run: docker load < /tmp/webapp.tar
+      ##########################################################################
+      # UNIT TESTS WEBAPP #####################################################
+      ##########################################################################
+      # TODO: Why do we need those .envs?
+      - name: backend | copy env files webapp
+        run: cp webapp/.env.template webapp/.env
+      - name: backend | copy env files backend
```
Comment (Mogge) [inline .github/workflows/ci.yml:238]: Same here. Does it work without the env of backend?

**Review by Mogge — COMMENTED** [review]
Wow. Great code. Easy to read because of the comments. I just have some questions, so feel free to merge.

**Review by Mogge — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ulfgebhardt) [reply .github/workflows/ci.yml:133]: I made a followup Issue to discuss this @Mogge https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4188

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ulfgebhardt) [reply .github/workflows/ci.yml:162]: I made a followup Issue to discuss this @Mogge https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4188

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ulfgebhardt) [reply .github/workflows/ci.yml:197]: Followup ticket: https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4189

File: .github/workflows/ci.yml (hunk 4 above)
Comment (ulfgebhardt) [reply .github/workflows/ci.yml:238]: Followup ticket: https://github.com/Ocelot-Social-Community/Ocelot-Social/issues/4189

**Pull request merged**